XWOpt/TextureBasisVectors.cs

```diff
@@ -0,0 +1,51 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+
+namespace SchmooTech.XWOpt
+{
+    public struct TextureBasisVectors<TVector3>
+    {
+        private TVector3 accrossTop;
+        private TVector3 downSide;
+
+        public TVector3 AcrossTop { get => accrossTop; set => accrossTop = value; }
+        [System.Diagnostics.CodeAnalysis.SuppressMessage("Microsoft.Naming", "CA1702:CompoundWordsShouldBeCasedCorrectly", MessageId = "DownSide")]
+        public TVector3 DownSide { get => downSide; set => downSide = value; }
+
+        internal TextureBasisVectors(OptReader reader)
+        {
+            accrossTop = reader.ReadVector<TVector3>();
+            downSide = reader.ReadVector<TVector3>();
+        }
+
+        public override bool Equals(object obj)
+        {
+            if (!(obj is TextureBasisVectors<TVector3>))
```

Comment:
var tvec = obj as TextureBasisVectors....

Do this like you have done in other files. That pattern is better.